api/src/main/java/io/minio/MinioClient.java

```diff
@@ -3927,15 +3927,52 @@ public void enableObjectLegalHold(String bucketName, String objectName, String v
       throws ErrorResponseException, IllegalArgumentException, InsufficientDataException,
           InternalException, InvalidBucketNameException, InvalidKeyException,
           InvalidResponseException, IOException, NoSuchAlgorithmException, XmlParserException {
+    this.enableObjectLegalHold(
+        EnableObjectLegalHoldArgs.builder()
+            .bucket(bucketName)
+            .object(objectName)
+            .versionId(versionId)
+            .build());
+  }
+
+  /**
+   * Enables legal hold on an object.
+   *
+   * <pre>Example:{@code
+   * minioClient.enableObjectLegalHold(
+   *    EnableObjectLegalHoldArgs.builder()
+   *        .bucket("my-bucketname")
+   *        .object("my-objectname")
+   *        .versionId("object-versionId")
+   *        .build());
+   * }</pre>
+   *
+   * @param args {@link EnableObjectLegalHoldArgs} object.
+   * @throws ErrorResponseException thrown to indicate S3 service returned an error response.
+   * @throws IllegalArgumentException throws to indicate invalid argument passed.
+   * @throws InsufficientDataException thrown to indicate not enough data available in InputStream.
+   * @throws InternalException thrown to indicate internal library error.
+   * @throws InvalidBucketNameException thrown to indicate invalid bucket name passed.
+   * @throws InvalidKeyException thrown to indicate missing of HMAC SHA-256 library.
+   * @throws InvalidResponseException thrown to indicate S3 service returned invalid or no error
+   *     response.
+   * @throws IOException thrown to indicate I/O error on S3 operation.
+   * @throws NoSuchAlgorithmException thrown to indicate missing of MD5 or SHA-256 digest library.
+   * @throws XmlParserException thrown to indicate XML parsing error.
+   */
+  public void enableObjectLegalHold(EnableObjectLegalHoldArgs args)
+      throws ErrorResponseException, IllegalArgumentException, InsufficientDataException,
+          InternalException, InvalidBucketNameException, InvalidKeyException,
+          InvalidResponseException, IOException, NoSuchAlgorithmException, XmlParserException {
     Map<String, String> queryParamMap = new HashMap<>();
     queryParamMap.put("legal-hold", "");
 
-    if (versionId != null && !versionId.isEmpty()) {
-      queryParamMap.put("versionId", versionId);
+    if (args.versionId() != null) {
```

Comment:
Was the condition `&& !versionId.isEmpty()` removed intentionally? Seems like a useful validation.
Might be a good idea to add it as a validation (must be non-empty if non-null) in the args builder.

Reply:
As AWS works with empty string as `versionId`, the check is OK